src/core/router/transports/ssu/packet.cc

```diff
@@ -893,7 +903,18 @@ std::unique_ptr<SSUPeerTestPacket> SSUPacketParser::ParsePeerTest() {
   auto packet = std::make_unique<SSUPeerTestPacket>();
   packet->SetNonce(Read<std::uint32_t>());
   std::uint8_t const size(Read<std::uint8_t>());
-  packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));
+  if (size)
+    {
+      if (size == SSUSize::AddressIPv4 || size == SSUSize::AddressIPv6)
+        {
+          packet->SetIPAddressSize(size);
+          packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));
+        }
+      else
+        throw std::length_error("SSU: invalid IP address size");
+    }
+  else
+    packet->SetIPAddressSize(size);
```

Comment:
I would advise something much simpler / less redundant:

```diff
diff --git a/src/core/router/transports/ssu/packet.cc b/src/core/router/transports/ssu/packet.cc
index ce0a7255..1d189479 100644
--- a/src/core/router/transports/ssu/packet.cc
+++ b/src/core/router/transports/ssu/packet.cc
@@ -903,18 +903,9 @@ std::unique_ptr<SSUPeerTestPacket> SSUPacketParser::ParsePeerTest() {
   auto packet = std::make_unique<SSUPeerTestPacket>();
   packet->SetNonce(Read<std::uint32_t>());
   std::uint8_t const size(Read<std::uint8_t>());
-  if (size)
-    {
-      if (size == SSUSize::AddressIPv4 || size == SSUSize::AddressIPv6)
-        {
-          packet->SetIPAddressSize(size);
-          packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));
-        }
-      else
-        throw std::length_error("SSU: invalid IP address size");
-    }
-  else
-    packet->SetIPAddressSize(size);
+  if (size)  // Bob or Charlie
+    packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));
+  packet->SetIPAddressSize(size);
   packet->SetPort(Read<std::uint16_t>());
   packet->SetIntroKey(ReadBytes(SSUSize::IntroKey));
   return packet;
diff --git a/src/core/router/transports/ssu/packet.h b/src/core/router/transports/ssu/packet.h
index 2af971c2..84f20b51 100644
--- a/src/core/router/transports/ssu/packet.h
+++ b/src/core/router/transports/ssu/packet.h
@@ -54,8 +54,6 @@ enum SSUSize : std::uint16_t
   PacketMaxIPv4 = MTUv4 - HeaderIPv4 - HeaderUDP,  // Total: 1456
   PacketMaxIPv6 = MTUv6 - HeaderIPv6 - HeaderUDP,  // Total: 1440
   HeaderMin = 37,
-  AddressIPv4 = 4,
-  AddressIPv6 = 16,
   MAC = 16,
   IV = 16,
   IntroKey = 32,
```

Invalid size will throw during conversion and we will catch appropriately.